tests/unit/test_server.py

```diff
@@ -94,3 +94,51 @@ def test_prometheus_server_with_route_prefix_reload_configuration_failure(self):
         )
 
         self.assertFalse(self.prometheus.reload_configuration())
+
+    @responses.activate
```

Comment:
Wouldn't it be more valuable to test that in response to a update status event the Prometheus charm sets blocked or unknown status when Prometheus server is not healthy or not ready respectively. 

This and the following unit tests only check rather trivial behaviors of a single if statement in two methods. Note I am not asking for removal of these unit tests. I am trying to say the more vital unit tests is missing.